lib/storage/src/dispatcher.rs

```diff
@@ -46,39 +45,11 @@ impl Dispatcher {
         // if distributed deployment is enabled
         if let Some(state) = self.consensus_state.as_ref() {
             let op = match operation {
-                CollectionMetaOperations::CreateCollection(mut op) => {
+                CollectionMetaOperations::CreateCollection(op) => {
+                    // TODO: move this check also?
                     self.toc.check_write_lock()?;
```

Comment:
Need your advice on this - should we also move this code from dispatcher to somewhere else?